golang/cosmos/x/vstorage/vstorage.go

```diff
@@ -57,46 +58,90 @@ func (sh vstorageHandler) Receive(cctx *vm.ControllerContext, str string) (ret s
 	// Handle generic paths.
 	switch msg.Method {
 	case "set":
-		keeper.SetStorageAndNotify(cctx.Context, msg.Path, msg.Value)
+		for _, arg := range msg.Args {
+			var entry types.StorageEntry
+			entry, err = types.UnmarshalStorageEntry(arg)
+			if err != nil {
+				return
+			}
+			keeper.SetStorageAndNotify(cctx.Context, entry)
+		}
```

Comment:
If I'm reading correctly, "set" and all the other write methods now affect an arbitrary number of entries. Would it instead make sense to separate single-entry vs. multi-entry methods?

Reply:
I didn't see a reason to have different methods for single vs multiple entry updates if the length of the args indicates that intent.